chainer/link.py

```diff
@@ -552,35 +646,57 @@ def __getitem__(self, name):
         """Equivalent to getattr."""
         return getattr(self, name)
 
+    def __setattr__(self, name, value):
```

Comment:
OK, it would be desirable if we add tests that ensures the correctness of backprop even though there are multiple registrations.